rtt_ros2_node/src/rtt_ros2_node.cpp

```diff
@@ -121,4 +128,45 @@ void Node::cancel()
   }
 }
 
+Node::shared_ptr getNodeService(RTT::TaskContext * tc)
+{
+  Node::shared_ptr node;
+
+  if (tc != nullptr) {
+    // Try to find the service by name
+    node = boost::dynamic_pointer_cast<Node>(tc->provides()->getService("Node"));
+    if (node != nullptr) {
+      return node;
+    }
+
+    // Try to find the service by type
+    for (const auto & name : tc->provides()->getProviderNames()) {
+      node = boost::dynamic_pointer_cast<Node>(tc->provides()->getService(name));
```

Comment:
Does this casting return `nullptr` when the service of the iterator is not of type `rtt_ros2_node::Node` ?
What a cool idea!